src/fmu/dataio/export/rms/structure_depth_fault_lines.py

```diff
@@ -0,0 +1,154 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Final
+
+import fmu.dataio as dio
+from fmu.dataio._logging import null_logger
+from fmu.dataio._models.fmu_results.enums import (
+    Classification,
+    Content,
+    StandardResultName,
+)
+from fmu.dataio._models.fmu_results.standard_result import (
+    StructureDepthFaultLinesStandardResult,
+)
+from fmu.dataio.exceptions import ValidationError
+from fmu.dataio.export._decorators import experimental
+from fmu.dataio.export._export_result import ExportResult, ExportResultItem
+from fmu.dataio.export.rms._utils import (
+    get_polygons_in_folder,
+    get_rms_project_units,
+    load_global_config,
+)
+
+if TYPE_CHECKING:
+    import xtgeo
+
+_logger: Final = null_logger(__name__)
+
+
+class _ExportStructureDepthFaultLines:
+    def __init__(
+        self,
+        project: Any,
+        horizon_folder: str,
+    ) -> None:
+        _logger.debug("Process data, establish state prior to export.")
+        self._horizon_folder = horizon_folder
+        self._config = load_global_config()
+        self._fault_lines = get_polygons_in_folder(project, horizon_folder)
+        self._unit = "m" if get_rms_project_units(project) == "metric" else "ft"
+
+        _logger.debug("Process data... DONE")
+
+    @property
+    def _standard_result(self) -> StructureDepthFaultLinesStandardResult:
+        """Product type for the exported data."""
+        return StructureDepthFaultLinesStandardResult(
+            name=StandardResultName.structure_depth_fault_lines
+        )
+
+    @property
+    def _classification(self) -> Classification:
+        """Get default classification."""
+        return Classification.internal
+
+    @property
+    def _subfolder(self) -> str:
+        """Subfolder for exporting the standard result to."""
+        return StandardResultName.structure_depth_fault_lines.value
+
+    def _export_fault_line(self, pol: xtgeo.Polygons) -> ExportResultItem:
+        edata = dio.ExportData(
+            config=self._config,
+            content=Content.fault_lines.value,
+            unit=self._unit,
+            vertical_domain="depth",
+            domain_reference="msl",
+            subfolder=self._subfolder,
+            is_prediction=True,
+            name=pol.name,
+            classification=self._classification,
+            rep_include=True,
+        )
+
+        edata.polygons_fformat = "parquet"  # type: ignore
```

Comment:
mypy doesn't like setting a class variable on an instance, hence the type ignore here.